### PR TITLE
Add the authorization Step for the Application Passwords

### PR DESCRIPTION
**Pull request by valterlorran**
Related to #

## Proposed Changes

Adds the new Authorization Step to allow us to capture the Application Password authorization. This step gives the user the possibility to authorize our application so we can use the credentials for the automate checks and tasks.

I also added the possibility of adding `notices` to the Step. I updated the StepContainer to accept a new prop: `notice`. And it is displayed before the Formatted Header.

### Default view

Captura de Tela 2024-12-03 às 20 23 56

### Rejected the Authorization

Captura de Tela 2024-12-03 às 20 23 43

### Error trying to store the Authorization credentials

Captura de Tela 2024-12-03 às 20 23 23

## Why are these changes being made?

* This is an attempt to improve the low success rates of users credentials. Most of the credentials that the users inputs are not valid and we can proceed with many automated processes that would save a great deal of time for the HEs. More context on:  paYKcK-5B6-p2

## Testing Instructions

_Visual inspection should be enough, as I added some tests to cover the cases._

To test it manually follow the steps:

* Use calypso live to avoid the SSL error when trying to authenticate.
* Go through the migration process until you reach the Credentials step.
* Then update the URL to point to the `/setup/site-migration/site-migration-application-password-authorization` path. Keep the current parameters, and add the `authorizationUrl`, which could be a valid authorization URL for the Application Passwords: `https://example.com/wp-admin/authorize-application.php&app_id=1083d914-d9f8-47ec-b49b-55d206104730&app_name=Test`
* You should see the default view, inviting the user to Authorize. Click on the "Authorize" button.
* You should be redirected to your test site.
* Reject the request, and you should be redirected to the Authorization step with an alert.
* Click again on the "Authorize" button, and this time approve the request.
* You should be redirected to the Authorization step and should see a loading state.
* After the loading state, you'll probably see the Authorization step with an error message, because the endpoint that stores the Password will fail because you don't have the ticket.
* So far this should be enough to test.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by andres-blanco — CHANGES_REQUESTED** [review]
Here's a bunch of small design differences that doesn't match the figma file:
* Spacing between the title and the paragraph below
* The site URL should be bold
* Spacing between paragraph and CTA button 
* The weight of the `Share credentials instead` link is different
* Spacing between `Share credentials instead` link and `Here's what else you're getting` title is different
* Spacing between `Here's what else you're getting` title and the description box is different
* The description box items have icons in the figma
* Font size of the description box items is different

Also, the `Need help? Let us guide you` link in the top right is missing.
image

Please tackle the easy fixes and leave the rest for a subsequent PR.

**Comment by andres-blanco** [comment]
@valterlorran the CTA en the Figma reads `Authorize access`, in the PR is `Authorize`

**Comment by andres-blanco** [comment]
https://github.com/user-attachments/assets/9ab761fc-7592-41fb-9e7b-f42f7122bef1

When I click authorize I get redirected to the first step of the flow. Based on the `undefined` that appears on the URL this might be related to the `site_url` vs `from` parameter

**Comment by valterlorran** [comment]
@andres-blanco 

>  Spacing between the title and the paragraph below

 We should not override these values to not create inconsistencies with other steps. We should use the default behavior of the component.
> The site URL should be bold

Fixed
> Spacing between paragraph and CTA button

 I’m assuming this means the space between the `We're ready to migrate…` subtitle and the `Authorize access` button: Again the space added between is default of the header and the step container, reducing that would be different from other steps.
> The weight of the Share credentials instead link is different

I’m using the default link that we have been using in other steps. I think if we decide to tackle that, we should do a refactor.
> Spacing between Share credentials instead link and Here's what else you're getting title is different

 Fixed
> Spacing between Here's what else you're getting title and the description box is different

Fixed
> The description box items have icons in the figma

 Isn’t that showing up for you? Can you share a screen shot?
> Font size of the description box items is different

I used the $font-body-small instead of hardcoding the site and to have consistence.

**Comment by andres-blanco** [comment]
> Isn’t that showing up for you? Can you share a screen shot?
image

Figma: F8foKv974Gc5dTHSh0i0Wo-fi-648_7452

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-application-password-authorization/components/authorization.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+					onClick={ onShareCredentialsClick }
+				>
+					{ translate( 'Share credentials instead' ) }
+				</button>
+			</div>
+			<div className="site-migration-application-password-authorization__benefits-container">
+				<h3>{ translate( "Here's what else you're getting" ) }</h3>
+				<AuthorizationBenefits
+					benefits={ [
+						translate( 'Uninterrupted service throughout the entire migration experience.' ),
```
Comment (andres-blanco) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-application-password-authorization/components/authorization.tsx:50]: where this strings added to the string freeze PR we where using for having translations ready earliear?

**Review by andres-blanco — CHANGES_REQUESTED** [review]
The code overall looks good and the happy path seems to be working correctly. I found a couple of blockers.

* When I get redirected to the wp-admin screen to authorize there's no Application Name pre-filled (we should user the `app_name` query param specified here https://make.wordpress.org/core/2020/11/05/application-passwords-integration-guide/)
image

* The let us guide you in the top right is still missing

**Comment by valterlorran** [comment]
> When I get redirected to the wp-admin screen to authorize there's no Application Name pre-filled (we should user the app_name query param specified here

This should be working, the thing is that on the testing instructions I didn't add the App ID and App Name, but this should come from the endpoint. In order for that to work just add the app_id and the app_name params to your authorizationUrl:

Please see: fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Szvtengvba%2Snhgbzngrq%2Qzvtengvba%2Qgbbyf.cuc%3Se%3Q9r16q365%231171%2Q1178-og

**Review by andres-blanco — APPROVED** [review]
Based on [this comment ](https://github.com/Automattic/wp-calypso/pull/96891#issuecomment-2518140902) I think this is in a good state for merging.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17046536).

**Pull request merged**